diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/StringConcatenationRule.kt

```diff
@@ -71,4 +88,63 @@ class StringConcatenationRule : Rule("string-concatenation") {
         return operationReference
                 ?.getFirstChildWithType(PLUS) != null
     }
+
+    private fun fixBinaryExpressionWithConcatenation(node: ASTNode?): ASTNode? {
+        return node?.let {
+            val binaryExpressionNode = node.psi as KtBinaryExpression
+
+            val parentNode = node.treeParent
+
+            // this is a special very tricky hack: we have special utility methods (KotlinParser)
+            // that let us to create a node from a string automatically by this utility
+            // instead of creating a node in AST manually by a programmer and creating an hierarchy of nodes in a tree
+
+            // FixMe: code duplication here is temporary until fix will work properly
+            if (binaryExpressionNode.isRvalueConstantExpression() || binaryExpressionNode.isRvalueStringTemplateExpression()) {
+                // =========== "a " + "b" -> "a b"
+                val lvalueText = binaryExpressionNode.left!!.text.replace("\"", "")
+                val rvalueText = binaryExpressionNode.right!!.text.replace("\"", "")
+                val newNode = KotlinParser().createNode("\"$lvalueText$rvalueText\"")
+                parentNode.replaceChild(node, newNode)
+                return newNode
+            } else if (binaryExpressionNode.isRvalueReferenceExpression()) {
+                // ===========  "a " + b -> "a $b"
+                val lvalueText = binaryExpressionNode.left!!.text.replace("\"", "")
+                val rvalueText = binaryExpressionNode.right!!.text
+                val newNode = KotlinParser().createNode("\"$lvalueText$$rvalueText\"")
+                parentNode.replaceChild(node, newNode)
+                return newNode
+            } else if (binaryExpressionNode.isRvalueDotQualifiedExpression()) {
+                // ===========  "string " + "valueStr".replace("my", "") -> "string ${"valueStr".replace("my", "")}""
+                val lvalueText = binaryExpressionNode.left!!.text.replace("\"", "")
+                val rvalueText = binaryExpressionNode.right!!.text
+                val newNode = KotlinParser().createNode("\"$lvalueText\${$rvalueText}\"")
+                parentNode.replaceChild(node, newNode)
+                return newNode
+            } else if (binaryExpressionNode.isRvalueParenthesized()) {
+                // ===========  "string " + (1 + 2) -> "string ${(1 + 2)}"
```

Comment:
Seems like parenthesis can be safely removed here, or do you think it would be to aggressive fixing?